lib/opener-registry.js

```diff
@@ -0,0 +1,56 @@
+/** @babel */
+
+import _ from 'lodash'
+import fs from 'fs-plus'
+import path from 'path'
+
+export default class OpenerRegistry {
+  constructor () {
+    this.createOpeners()
+  }
+
+  async createOpeners () {
+    const moduleDir = path.join(__dirname, 'openers')
+    const entries = fs.readdirSync(moduleDir)
+    this.openers = entries.map(entry => {
+      const OpenerImpl = require(path.join(moduleDir, entry))
+      return new OpenerImpl()
+    })
+    // Sort the openers into decreasing priority
+    this.openers = _.orderBy(this.openers, [opener => opener.hasSynctex(), opener => opener.canOpenInBackground()], ['desc', 'desc'])
+  }
+
+  async open (filePath, texPath, lineNumber) {
+    const name = atom.config.get('latex.opener')
+    const openResultInBackground = atom.config.get('latex.openResultInBackground')
+    const enableSynctex = atom.config.get('latex.enableSynctex')
+
+    let openers = _.filter(this.openers, opener => opener.canOpen(filePath))
```

Comment:
`const openers = this.openers.filter(opener => opener.canOpen(filePath))`

Also, I'd probably name it `candidates` instead of `openers` to indicate this is the list of candidates from which we will select an opener.